ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/restore/KsqlRestoreCommandTopic.java

```diff
@@ -287,13 +288,14 @@ public static void main(final String[] args) throws Exception {
   }
 
   KsqlRestoreCommandTopic(final KsqlConfig serverConfig) {
-    this(
-        serverConfig,
-        ReservedInternalTopics.commandTopic(serverConfig),
-        ServiceContextFactory.create(serverConfig,
-            () -> /* no ksql client */ null).getTopicClient(),
-        () -> transactionalProducer(serverConfig)
-    );
+    final Map<String, Object> adminClientConfigs =
+      new HashMap<>(serverConfig.getKsqlAdminClientConfigProps());
+    adminClientConfigs.putAll(serverConfig.originalsWithPrefix("ksql.server.command.consumer."));
```

Comment:
Can you still keep the reference to the "this" constructor? You can create a helper function that does these couple lines
```
final Map<String, Object> adminClientConfigs =
      new HashMap<>(serverConfig.getKsqlAdminClientConfigProps());
  adminClientConfigs.putAll(serverConfig.originalsWithPrefix("ksql.server.command.consumer."));
final Admin admin = new DefaultKafkaClientSupplier().getAdmin(adminClientConfigs);
```
and returns an Admin object
```
new KafkaTopicClientImpl(() -> createAdminClient(serverConfig));
```

Reply:
I'm not sure I can - when I tried intellij gives me a `Cannot reference 'KsqlRestoreCommandTopic.createAdminClient' before supertype constructor has been called`, it seems like you can't call anything if you're calling the second constructor

Reply:
You have to make `createAdminClient` and it should work

Reply:
aha, that's the trick, updated